module/os/linux/zfs/zvol_os.c

```diff
@@ -1570,8 +1593,40 @@ zvol_init(void)
 		zvol_actual_threads = MIN(MAX(zvol_threads, 1), 1024);
 	}
 
+	/*
+	 * Use atleast 32 zvol_threads but for many core system,
```

Comment:
I think we can simplify to make it easer to reason about. One idea is:
- Systems with >32 CPUs retain previous behavior of 1 taskq and 32 threads or zvol_threads value
- Systems with 32 or greater will scale the number of taskq with CPU count / [6|8] (or some other #define value).  And per taskq thread count can be 8 (perhaps making this a tunable)

It would be close to the values and behavior you have and the code would be a little simpler.

Reply:
Apologies if I misunderstood your reasoning, but it's worth noting that single taskq lock contention can be observed even on systems with fewer than 32 CPUs.